src/server.c

```diff
@@ -136,7 +136,8 @@ void serverLogRaw(int level, const char *msg) {
         struct tm tm;
         nolocks_localtime(&tm,tv.tv_sec,server.timezone,server.daylight_active);
         off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
-        snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+        time_t zone = server.timezone / 3600;
+        snprintf(buf+off,sizeof(buf)-off,"%03d %c%02ld00",(int)tv.tv_usec/1000,zone>0?'-':'+',labs(zone));
```

Comment:
You can replace that "complex" logic for calculating the offset using another call to `strftime` using `%z` : 

```suggestion
        off += snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
        strftime(buf+off,sizeof(buf),"%z",&tm);
```

The linux man page for [`strftime`](https://man7.org/linux/man-pages/man3/strftime.3.html) documents `%z`:
>       %z     The +hhmm or -hhmm numeric timezone (that is, the hour and
>              minute offset from UTC). (SU)

unfortunately that will produce the less desirable but equally correct `+0100` (no colon) instead of `+01:00` (colon between hours and minutes)

Reply:
Strangely `strftime` does not work on my centOS 7 machine... and that is why I did not use it. It does work on my macOS though. I will try to figure out what's going on.